CST338/Seperate files/Module4/DataMatrix.java

```diff
@@ -57,18 +57,41 @@ public boolean scan(BarcodeImage bc) {
    
    public boolean generateImageFromText() {
       int asciiValue;
+      int asteriskASCII = 42;
       int index = 0;
 
       clearImage();
-      actualHeight = 10; // 8 bits plus bottom and top borders
-      actualWidth = text.length() + 2; // + 2 for left and right borders
-
+      actualHeight = 10; // 8 bits plus bottom and top spline
+      actualWidth = text.length() + 2; // + 2 for left and right spline
+      // write top spline row to image
+      
+      for (int i = 0; i < actualWidth; i++) {
+         if (i % 2 == 0) {
+            image.setPixel(0,i,true);
+         }
+         else {
+            image.setPixel(0,i,false);
+         }
```

Comment:
If branch only has one line of code, then it looks cleaner as
if (i % 2 == 0) {
   image.setPixel(0, i, true);
else
   image.setPixel(0, i, false);
}